module-gui/gui/widgets/text/Text.cpp

```diff
@@ -445,10 +451,15 @@ namespace gui
         drawLines();
     }
 
+    TextCursor *Text::createCursor()
+    {
+        return textType == TextType::MultiLine ? new TextLineCursor(this) : new TextCursor(this);
+    }
+
     void Text::buildCursor()
     {
         erase(cursor);
-        cursor = new TextLineCursor(this);
+        cursor = createCursor();
```

Comment:
this and next `if` points in the direction that we should either:
- make buildCursor virtual in Text, and implement it in specializations (multiline & singleline & whateverline)
- use GUI::Text as a universal facade and move this implementation into two classes that handles manipulations (one for multiline, one for single line) this was the way the code was going for some time

Reply:
To be discussed.